app.go

```diff
@@ -0,0 +1,423 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/gorilla/mux"
+	"net/http"
+	"time"
+	"fmt"
+	"io"
+	"io/ioutil"
+)
+
+type AsyncTasksApp struct {
+	db     *DBConnection
+	router *mux.Router
+}
+
+func NewAsyncTasksApp(db *DBConnection, router *mux.Router) *AsyncTasksApp {
+	app := &AsyncTasksApp{
+		db:     db,
+		router: router,
+	}
+
+	app.InitRoutes()
+
+	return app
+}
+
+func (a *AsyncTasksApp) InitRoutes() {
+	a.router.NotFoundHandler = http.HandlerFunc(a.NotFound)
+	a.router.HandleFunc("/tasks/{id:[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}}", a.GetByIdRequest).Methods("GET").Name("getById")
+	a.router.HandleFunc("/tasks/{id:[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}}", a.DeleteByIdRequest).Methods("DELETE").Name("deleteById")
+	a.router.HandleFunc("/tasks/{id:[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}}/status", a.AddStatusRequest).Methods("POST").Name("addStatus")
+	a.router.HandleFunc("/tasks/{id:[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}}/behaviors", a.AddBehaviorRequest).Methods("POST").Name("addBehavior")
+
+	a.router.HandleFunc("/tasks", a.GetByFilterRequest).Methods("GET").Name("getByFilter")
+	a.router.HandleFunc("/tasks", a.CreateTaskRequest).Methods("POST").Name("createTask")
+}
+
+func (a *AsyncTasksApp) NotFound(writer http.ResponseWriter, r *http.Request) {
+	notFound(writer, fmt.Sprintf("no endpoint found at %s %s", r.Method, r.URL.Path))
+}
+
+func (a *AsyncTasksApp) GetByIdRequest(writer http.ResponseWriter, r *http.Request) {
```

Comment:
It might make sense to split out the handler functions into another file/other files. But this is where I started with them, so for now here they are.